iterator.js

```diff
@@ -52,32 +53,40 @@ util.inherits(Iterator, AbstractIterator)
 Iterator.prototype.createIterator = function() {
   var self = this
 
-  self.iterator = self.db.iterate(function () {
+  self.transaction = self.db.iterate(function () {
     self.onItem.apply(self, arguments)
   }, {
     keyRange: self._keyRange,
     autoContinue: false,
     order: self._order,
-    onError: function(err) {
-      if (err.type !== 'abort' && !self._ended) {
-        // TODO: pass to next() callback
-        console.error('horrible error', err)
-      }
-    }
+
+    // If an error occurs, the transaction will abort and we can
+    // get the error from "transaction.error".
+    onError: noop
   })
+
+  // Override IDBWrapper's event handlers for a simpler flow.
+  self.transaction.oncomplete = self.transaction.onabort = function () {
+    self.onComplete()
```

Comment:
:+1:

Reply:
Or `= self.onComplete.bind(self)` but nit pick

Reply:
I figured it's more convenient to override if we want to unit test it